app/bundles/CoreBundle/EventListener/CoreSubscriber.php

```diff
@@ -16,72 +16,113 @@
 use Mautic\CoreBundle\Menu\MenuHelper;
 use Mautic\CoreBundle\Templating\Helper\AssetsHelper;
+use Mautic\FormBundle\Entity\FormRepository;
 use Mautic\InstallBundle\Controller\InstallController;
 use Mautic\UserBundle\Entity\User;
 use Mautic\UserBundle\Event\LoginEvent;
 use Mautic\UserBundle\Model\UserModel;
 use Mautic\UserBundle\UserEvents;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
 use Symfony\Component\Routing\Route;
 use Symfony\Component\Routing\RouteCollection;
-use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Core\SecurityContext;
 use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
 use Symfony\Component\Security\Http\SecurityEvents;
+use Symfony\Component\Translation\TranslatorInterface;
 
-/**
- * Class CoreSubscriber.
- */
-class CoreSubscriber extends CommonSubscriber
+class CoreSubscriber implements EventSubscriberInterface
 {
     /**
      * @var BundleHelper
      */
-    protected $bundleHelper;
+    private $bundleHelper;
 
     /**
      * @var MenuHelper
      */
-    protected $menuHelper;
+    private $menuHelper;
 
     /**
      * @var UserHelper
      */
-    protected $userHelper;
+    private $userHelper;
 
     /**
      * @var AssetsHelper
      */
-    protected $assetsHelper;
+    private $assetsHelper;
 
     /**
      * @var SecurityContext
      */
-    protected $securityContext;
+    private $securityContext;
 
     /**
      * @var UserModel
      */
-    protected $userModel;
+    private $userModel;
 
     /**
      * @var CoreParametersHelper
      */
-    protected $coreParametersHelper;
+    private $coreParametersHelper;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $dispatcher;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var FormRepository
+     */
+    private $formRepository;
+
+    /**
+     * System parameters.
+     *
+     * @var array
+     */
+    private $params;
+
+    /**
+     * @var MauticFactory
+     */
+    private $factory;
 
     public function __construct(
         BundleHelper $bundleHelper,
         MenuHelper $menuHelper,
         UserHelper $userHelper,
         AssetsHelper $assetsHelper,
         CoreParametersHelper $coreParametersHelper,
-        AuthorizationCheckerInterface $securityContext,
-        UserModel $userModel
+        SecurityContext $securityContext,
+        UserModel $userModel,
+        EventDispatcherInterface $dispatcher,
+        TranslatorInterface $translator,
+        RequestStack $requestStack,
+        FormRepository $formRepository,
+        array $params,
```

Comment:
Do you know why we're using the params array rather than the `CoreParametersHelper` here?

Reply:
I'm using the same dependencies as the CommonSubscriber provided for the classes that extended it. I do not want to risk introducing bugs changing code that is not necessary to reach the goal.